pagefaultgames/pokerogue
# [QoL] Add ALL option when transferring items to Pokemons

## What are the changes?

This PR introduces the "ALL" option in the transfer selection dropdown when transferring items from one Pokemon to another.

## Why am I doing these changes?



I recently was playing in the Endless mode, and started to build a "mule", that would carry items until I find a good Pokemon to transfer them to. I realized that I had to click multiple times to transfer all items from one Pokemon to another, and thought that it would be a good idea to have an "ALL" option in the dropdown.

## What did change?



From a user experience, the dropdown now has an "ALL" option, that will transfer all items from the source Pokemon to the target Pokemon. This takes the limitations of the current transfer system into account, where you cannot transfer to a Pokemon that has already reached the maximum number of a specific item.

Code-wise, I've added a new attribute to the `PartyUiHandler` to track whether or not we need to transfer all items.

I also took the liberty to add a translation file for `PartyUiHandler`, currently only containing the translation for the "ALL" option for `en`.

### Screenshots/Videos



- Before

  https://github.com/pagefaultgames/pokerogue/assets/57403591/3b86db61-ecd5-4ad9-aa13-ba1352804566


- After
  - When we can transfer all items

    https://github.com/pagefaultgames/pokerogue/assets/57403591/0fdec769-7135-4f9d-8434-7f320e1d3a96

  - When the receive already has the maximum number of items

    https://github.com/pagefaultgames/pokerogue/assets/57403591/f8183ff0-a64b-4317-9bd6-fc451c531a67

## How to test the changes?




In order to test the changes, you can modify the `src/overrides.ts` file to include the following code:

```ts
export const STARTING_LEVEL_OVERRIDE: integer = 100; // Just to be quicker to beat the first Pokemon
// ...
export const STARTING_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [
  { name: "LUCKY_EGG" },
  { name: "MULTI_LENS", count: 3 }
];
```

Then, you can start a new game with 2 Pokemon, and try to transfer items from one Pokemon to another. You should see the "ALL" option in the dropdown whenever there's more than one item to transfer.

To try the case where the target Pokemon already has the maximum number of items, you can modify the count of the `MULTI_LENS` item to be 5, and try to transfer items from the source Pokemon to the target Pokemon, since the maximum number of `MULTI_LENS` is 3 per Pokemon.

## Checklist
